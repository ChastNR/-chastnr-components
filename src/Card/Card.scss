.cd {
  flex: 0 0 50%;
  max-width: 50%;
  padding: 0 20px;
  margin-bottom: 0;

  &.small {
    flex: 0 0 33%;
  }

  .cd__box {
    height: 100%;
    transition: transform 0.3s, box-shadow 0.3s;
    transition-timing-function: ease;
    background: #fff;
    box-shadow: 0 4px 28px 0 rgba(0, 0, 0, 0.12);
    position: relative;
    display: block;
    overflow: hidden;
    width: 100%;
    color: #333;

    .cd__pctr {
      height: 144px;
      position: relative;
      background: linear-gradient(60deg, #f2f3f5, #d0d3d7);

      img {
        position: absolute;
        top: 50%;
        left: 50%;
        display: block;
        min-width: 64px;
        min-height: 64px;
        max-height: 100%;
        transform: translate3d(-50%, -50%, 0);
      }
    }

    .cd__cnt {
      padding: 32px;
      display: flex;
      flex-direction: column;
      flex-grow: 1;

      .cd__stn {
        margin-top: auto;
        display: flex;
        justify-content: space-between;
        align-items: center;

        & ~ .cd__stn {
          padding-top: 24px;
        }
      }
    }

    &.animate {
      &:hover {
        box-shadow: 0 12px 36px 0 rgba(0, 0, 0, 0.2);
        transform: translateY(-4px);
      }
    }

    & > div {
      display: flex;
      height: 100%;
      flex-direction: column;
    }
  }

  & > div {
    padding: 16px 0;
    display: flex;
    height: 100%;
  }

  @media (max-width: 500px) {
    flex: auto;
    max-width: 100%;
  }
}
