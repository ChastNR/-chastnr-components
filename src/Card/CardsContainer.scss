.cds__cnt {
  width: 100%;
  margin: 0 auto;
  display: flex;
  flex-wrap: wrap;
  flex-shrink: 0;
  flex-grow: 1;
  flex-basis: 0;
}
