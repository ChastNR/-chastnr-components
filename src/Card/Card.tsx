import { PropsWithChildren } from 'react';
import classNames from 'clsx';
import { Button } from '../Button';

import './Card.scss';

interface ICardProps
  extends PropsWithChildren<{
    animate?: boolean;
    className?: string;
    size?: 'small' | 'medium';
    imgSrc: string;
    buttonText: React.ReactNode;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
  }> {}

export const Card: React.FC<ICardProps> = ({
  animate = true,
  size = 'medium',
  imgSrc,
  buttonText,
  children,
  onClick,
}) => {
  const cardClassNames = classNames('cd', {
    size,
  });

  const boxClassNames = classNames('cd__box', {
    animate,
  });

  return (
    <div className={cardClassNames}>
      <div>
        <div className={boxClassNames}>
          <div>
            <div className="cd__pctr">
              <img alt="alt" src={imgSrc} />
            </div>
            <div className="cd__cnt">
              <div className="cd__stn">{children}</div>
              <div className="cd__stn">
                <Button onClick={onClick}>{buttonText}</Button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
