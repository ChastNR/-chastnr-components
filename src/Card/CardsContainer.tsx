import "./CardsContainer.scss";

const CardsContainer: React.FC = ({ children }) => <div className="cds__cnt">{children}</div>;

export default CardsContainer;
