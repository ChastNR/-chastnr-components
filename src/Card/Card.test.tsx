import { render } from '@testing-library/react';
import { Card } from './Card';

describe('Card', () => {
  it('should render properly', () => {
    const card = render(
      <Card buttonText="Click me" imgSrc="testImgSource">
        Text content
      </Card>
    );

    expect(card.container.firstChild).toMatchSnapshot();
  });
});
