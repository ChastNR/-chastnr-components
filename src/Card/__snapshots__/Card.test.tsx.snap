// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly 1`] = `
<div
  class="cd size"
>
  <div>
    <div
      class="cd__box animate"
    >
      <div>
        <div
          class="cd__pctr"
        >
          <img
            alt="alt"
            src="testImgSource"
          />
        </div>
        <div
          class="cd__cnt"
        >
          <div
            class="cd__stn"
          >
            Text content
          </div>
          <div
            class="cd__stn"
          >
            <button
              class="btn"
              type="button"
            >
              <div
                class="btn__txt__cntr"
                tabindex="-1"
              >
                <div
                  class="btn__txt__cntr__txt"
                >
                  Click me
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
